syntax = "proto3";
package pulse.study.bind.hydrocephalus;
option java_package = "pulse.study.bind.hydrocephalus";
option csharp_namespace = "pulse.study.bind.hydrocephalus";
option optimize_for = SPEED;

message SimulationData
{
  // Inputs
  // A filename without an extension (can/should have path also)
  // The engine will use this to create multiple output files (*.log and *.csv)
  uint32                       ID                                            = 1;
  string                       Name                                          = 2;
  double                       IntracranialSpaceVolume_mL                    = 3;
  double                       IntracranialSpaceCompliance_mL_Per_mmHg       = 4;
  double                       CerebrospinalFluidAbsorptionRate_mL_Per_min   = 5;
  double                       CerebrospinalFluidProductionRate_mL_Per_min   = 6;
  
  // Outputs
  double                       MeanBrainPressure_mmHg                        = 7;
  double                       MeanBrainVasculatureInFlow_mL_Per_s           = 8;
  double                       MeanBrainVasculatureOutFlow_mL_Per_s          = 9;
  double                       MeanBrainCarbonDioxidePartialPressure_mmHg    = 10;
  double                       MeanBrainOxygenPartialPressure_mmHg           = 11;
  // Some other system variable
  double                       CardiacOutput_mL_Per_min                      = 12;
  double                       CerebralBloodFlow_mL_Per_min                  = 13;
  double                       CerebralPerfusionPressure_mmHg                = 14;
  double                       DiastolicArterialPressure_mmHg                = 15;
  double                       HeartEjectionFraction                         = 16;
  double                       HeartRate_bpm                                 = 17;
  double                       HeartStrokeVolume_mL                          = 18;
  double                       IntracranialPressure_mmHg                     = 19;
  double                       MeanArterialPressure_mmHg                     = 20;
  double                       MeanCentralVenousPressure_mmHg                = 21;
  double                       PulmonaryDiastolicArterialPressure_mmHg       = 22;
  double                       PulmonaryMeanArterialPressure_mmHg            = 23;
  double                       PulmonaryMeanCapillaryFlow_mL_Per_s           = 24;
  double                       PulmonarySystolicArterialPressure_mmHg        = 25;
  double                       SystolicArterialPressure_mmHg                 = 26;
  double                       SystemicVascularResistance_mmHg_s_Per_L       = 27;

  // Stabilization Information
  bool                         AchievedStabilization                         = 28;
  double                       StabilizationTime_s                           = 29;
  double                       TotalSimulationTime_s                         = 30;
  
}

message SimulationListData
{
  string                       OutputRootDir                                 = 1;
  repeated SimulationData      Simulation                                    = 2;
}
